mne/epochs.py

```diff
@@ -443,6 +446,11 @@ def __init__(self, info, data, events, event_id=None, tmin=-0.2, tmax=0.5,
                 metadata = [metadata[s] for s in sub]
             elif metadata is not None:
                 metadata = metadata.iloc[sub]
+
+            # Remove temporarily set metadata from above, and set
+            # again to get the correct log ("adding metadata", instead of
+            # "replacing existing metadata")
+            GetEpochsMixin.metadata.fset(self, None, verbose=False)
```

Comment:
@larsoner [comment taken from here](https://github.com/mne-tools/mne-python/issues/10106#issuecomment-990035138)

> Or add a msg kwarg to the setter so you can say Adding instead of Replacing

I can do that if you absolutely hate the current solution :-) Let's discuss in this PR